torch/csrc/distributed/rpc/init.cpp

```diff
@@ -88,6 +92,13 @@ PyObject* rpc_init(PyObject* /* unused */) {
               [&](FutureMessage& fut) { return toPyObj(fut.wait()); },
               py::call_guard<py::gil_scoped_release>());
 
+  py::class_<ProcessGroupRpcAgentOptions>(
```

Comment:
docs for this one as well (can be done in separate PR)

Reply:
Any reason this is not using `shared_ptr_class_`?

Reply:
Didn't do it on purpose. What is the benefit of using `shared_ptr_class_ `?

Reply:
From [pybind doc](https://pybind11.readthedocs.io/en/stable/advanced/smart_ptrs.html):


> If no such holder type template argument is given, the default for a type named Type is std::unique_ptr<Type>, which means that the object is deallocated when Python’s reference count goes to zero.